cpp/src/arrow/io/file_test.cc

```diff
@@ -364,6 +365,18 @@ TEST_F(TestReadableFile, ReadAt) {
   ASSERT_RAISES(Invalid, file_->ReadAt(0, 1));
 }
 
+TEST_F(TestReadableFile, ReadAsync) {
+  MakeTestFile();
+  OpenFile();
+
+  ASSERT_OK_AND_ASSIGN(auto fut1, file_->ReadAsync(1, 10));
+  ASSERT_OK_AND_ASSIGN(auto fut2, file_->ReadAsync(0, 4));
+  ASSERT_OK_AND_ASSIGN(auto buf1, fut1.result());
+  ASSERT_OK_AND_ASSIGN(auto buf2, fut2.result());
```

Comment:
If `ReadAsync` returns a Future always then we can write `file_->ReadAsync(...).result()`